### PR TITLE
Fix server to look after Unstaked event before sending balance

### PR DESCRIPTION
**Pull request by shahafn**
Fixing #183

### CODE REVIEW

File: test/relay_client_test.js [hunk 1]
```diff
@@ -450,26 +451,49 @@ contract('RelayClient', function (accounts) {
...
-            relayBalance = await web3.eth.getBalance(relayServerAddress);
-            i++
-        }
-        assert.equal(0,relayBalance)
-        let afterOwnerBalance = await web3.eth.getBalance(relayOwner);
-        assert.equal(true,parseInt(afterOwnerBalance)  > parseInt(beforeOwnerBalance))
+    describe("relay balance management", async function () {
+        let relayServerAddress;
+        let beforeOwnerBalance;
+        it("should NOT send relay balance to owner after removed", async function () {
```
Comment (drortirosh) [inline test/relay_client_test.js]: Essentially, you wait here 4 seconds to verify the relay does not send out it's balance...

Can't you test it with go unittest?

File: test/relay_client_test.js [hunk 2]
```diff
@@ -450,26 +451,49 @@ contract('RelayClient', function (accounts) {
...
+
+            let i = 0;
+            let relayBalance = await web3.eth.getBalance(relayServerAddress);
+            while (relayBalance != 0 && i < 20) {
+                await testutils.sleep(200);
+                relayBalance = await web3.eth.getBalance(relayServerAddress);
+                i++
+            }
+            let afterOwnerBalance = await web3.eth.getBalance(relayOwner);
+            assert.equal(false,parseInt(afterOwnerBalance)  > parseInt(beforeOwnerBalance))
```
Comment (drortirosh) [inline test/relay_client_test.js]: Huh? Either do assert.ok, our (better) asset.equal.
Not really sure why you test it this way: the relay don't do any operation in this period, so you should do assert equal (before, after)

Comment (shahafn) [reply test/relay_client_test.js]: Yes I'm tired

File: test/relay_client_test.js (hunk 1 above)
Comment (shahafn) [reply test/relay_client_test.js]: I can, but this is not really a unit test, so decided to keep it in js.

File: test/relay_client_test.js [hunk 3]
```diff
@@ -450,26 +451,50 @@ contract('RelayClient', function (accounts) {
...
+            relayServerAddress = JSON.parse(response.body).RelayServerAddress;
+            beforeOwnerBalance = await web3.eth.getBalance(relayOwner);
+            let res = await rhub.removeRelayByOwner(relayServerAddress, {from:relayOwner});
+            let etherSpentByTx = res.receipt.gasUsed * (await web3.eth.getGasPrice());
+            assert.equal("RelayRemoved", res.logs[0].event);
+            assert.equal(relayServerAddress.toLowerCase(), res.logs[0].args.relay.toLowerCase());
+
+            let i = 0;
+            let relayBalance = await web3.eth.getBalance(relayServerAddress);
+            while (relayBalance != 0 && i < 10) {
```
Comment (drortirosh) [inline test/relay_client_test.js]: this loop is compeltely redundant here. simply do sleep(2000), and read the balance afterwards. 
You're not looking for the *shortest* time until it changes - you verify it doesn't change at all after 2 secoinds...

Comment (drortirosh) [reply test/relay_client_test.js]: and don't force-push changes. I can't see what changes you've made after my comments

File: test/relay_client_test.js (hunk 1 above)
Comment (alex-forshtat-tbk) [reply test/relay_client_test.js]: I am not sure js tests should cover server behavior.

Comment (shahafn) [reply test/relay_client_test.js]: This was already there. I think that as this one is more like integration tests so it's not the end of the world having these tests in js.

**Pull request merged**